### PR TITLE
refactor: Not pass raw types around in docmosis (FPLA-1856)

### PR DESCRIPTION
**Pull request by KiranVarma02**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-1856

### Change description ###
Not pass raw types around in docmosis

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ *] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/docmosis/GeneratedOrderTemplateDataGeneration.java [hunk 1]
```diff
@@ -40,22 +39,23 @@
...
+    abstract DocmosisGeneratedOrder populateCustomOrderFields(CaseData caseData);
 
     @Override
     public DocmosisGeneratedOrder getTemplateData(CaseData caseData) {
         OrderTypeAndDocument orderTypeAndDocument = caseData.getOrderTypeAndDocument();
         GeneratedOrderType orderType = orderTypeAndDocument.getType();
 
-        DocmosisGeneratedOrderBuilder<?, ?> orderBuilder = populateCustomOrderFields(caseData);
+        DocmosisGeneratedOrder docmosisGeneratedOrder = populateCustomOrderFields(caseData);
+        var docmosisGeneratedOrderBuilder = docmosisGeneratedOrder.toBuilder();
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/docmosis/GeneratedOrderTemplateDataGeneration.java:51]: More minor things but I think that the names `order` and `orderBuilder` are slightly better here. Not sure if it is the best practice to name variables after their type, sometimes it makes sense to like with case data but this time I am not as sure.

Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/docmosis/GeneratedOrderTemplateDataGeneration.java:51]: Could maybe merge these 2 lines as well. i.e.
``` java
var orderBuilder = populateCustomOrderFields(caseData).toBuilder();
```

**Review by lewisbirks — APPROVED** [review]
I am happy with this approach as long as we don't mind the very slight overhead of having to recreate the objects (but seeing as it is only once per document generation I don't think it is too bad)

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/docmosis/GeneratedOrderTemplateDataGeneration.java (hunk 1 above)
Comment (KiranVarma02) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/docmosis/GeneratedOrderTemplateDataGeneration.java:51]: there is another orderBuilder in code base. I may leave it like this `docmosisGeneratedOrderBuilder` , what do you think? but i combined 2 lines as suggested.

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/docmosis/AbstractOrderGenerationServiceTest.java [hunk 2]
```diff
@@ -57,23 +57,24 @@
             ));
     }
 
-    DocmosisGeneratedOrderBuilder<?, ?> enrichWithStandardData(GeneratedOrderType type,
+    DocmosisGeneratedOrder enrichWithStandardData(GeneratedOrderType type,
                                                                OrderStatus orderStatus,
-                                                               DocmosisGeneratedOrderBuilder orderBuilder) {
+                                                               DocmosisGeneratedOrder orderBuilder) {
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/docmosis/AbstractOrderGenerationServiceTest.java:62]: done

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/docmosis/AbstractOrderGenerationServiceTest.java [hunk 3]
```diff
@@ -57,23 +57,24 @@
...
         return enrichWithStandardData(type, null, orderStatus, orderBuilder);
     }
 
-    DocmosisGeneratedOrderBuilder<?, ?> enrichWithStandardData(GeneratedOrderType type,
+    DocmosisGeneratedOrder enrichWithStandardData(GeneratedOrderType type,
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/docmosis/AbstractOrderGenerationServiceTest.java:66]: done

**Review by lewisbirks — APPROVED** [review]

**Pull request merged**